go/mysql/datetime/parse.go

```diff
@@ -166,24 +166,35 @@ func clampTimeparts(tp *timeparts) bool {
 	if tp.day > 34 || tp.day == 34 && tp.hour > 22 {
 		tp.day = 34
 		tp.hour = 22
-		tp.min = 59
-		tp.sec = 59
+		tp.min = max(tp.min, 59)
```

Comment:
We should not just set `min` and `sec` to 59, because this makes parsed `min` and `sec` valid, even if these were invalid.

Reply:
@beingnoble03 I'm not sure I understand this one. This case is for when it overflows a `TIME` range and we then want to return the maximum possible value. That it's invalid is already indicated with the `return false` below. 

Do you have a specific `TIME` value here where this goes wrong at the moment? It feels like maybe you're fixing a bug in the wrong place here.

Reply:
@dbussink if we execute this query `SELECT TIME('11116656');`. `parsetimeAny` sets the `tp` (timeparts which we use to construct Time in `ParseTime`) as `1111:66:56` (which is invalid), and now as this is an overflow condition so`clampTimeparts` sets the `tp` as `838:59:59`, making it valid. But, this is wrong as MySQL returns `NULL` (because `1111:66:56` is invalid).

I have pushed the testcase now.

Reply:
@beingnoble03 Right, I do think that's a bug indeed, but I'm not sure if the fix for this is in the right place. It all seems to be becoming too complicated and I wonder if there's a simpler fix possible. I'll try to think about this to see if we're missing something and if this can be done in a better way.

Reply:
This is also with the special case zero time handling, that doesn't feel like it should be solved like that either tbh and that there must be some better way.

Reply:
I have opened https://github.com/vitessio/vitess/pull/15135 which I think is much more explicit about how things should work here. @beingnoble03 What do you think?